mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -0,0 +1,363 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail.task;
+
+import static org.apache.james.mailbox.cassandra.mail.migration.MailboxPathV2Migration.MAILBOX_PATH_V_2_MIGRATION_PERFORMED_VERSION;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.versions.CassandraSchemaVersionDAO;
+import org.apache.james.backends.cassandra.versions.SchemaVersion;
+import org.apache.james.mailbox.cassandra.ids.CassandraId;
+import org.apache.james.mailbox.cassandra.mail.CassandraIdAndPath;
+import org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMailboxPathV2DAO;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.task.Task;
+import org.apache.james.task.Task.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class SolveMailboxInconsistenciesService {
+    public static final Logger LOGGER = LoggerFactory.getLogger(SolveMailboxInconsistenciesService.class);
+
+    interface Inconsistency {
+        Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO);
+    }
+
+    private static class NoInconsistency implements Inconsistency{
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return Mono.just(Result.COMPLETED);
+        }
+    }
+
+    /**
+     * The Mailbox is referenced in MailboxDAO but the corresponding
+     * reference is missing in MailboxPathDao.
+     *
+     * In order to solve this inconsistency, we can simple re-reference the mailboxPath.
+     */
+    private static class OrphanMailboxDAOEntry implements Inconsistency {
+        private final Mailbox mailbox;
+
+        private OrphanMailboxDAOEntry(Mailbox mailbox) {
+            this.mailbox = mailbox;
+        }
+
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return pathV2DAO.save(mailbox.generateAssociatedPath(), (CassandraId) mailbox.getMailboxId())
+                .map(success -> {
+                    if (success) {
+                        LOGGER.info("Fixing inconsistency for orphan mailbox {} - {}",
```

Comment:
It looks like at this point, it's already done, the message can be misleading

Reply:
Should be `Inconsistency fixed for orphan mailbox {} - {}`

I'll investigate all similar messages to turn them in the past form.